server/automute.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/pkg/errors"
+)
+
+const (
+	NotifyPropAutomuted = "msteams_automute"
+
+	// This preference tracks if we've enabled automuting for the user by muting all of their connected channels.
+	PreferenceNameAutomuteEnabled = "automute_enabled"
+)
+
+// enableAutomute mutes all of the user's linked channels and DMs and sets a preference to track that automute is
+// enabled. It assumes that the caller has checked that the user is both connected and has their primary platform
+// set to MS Teams.
+func (p *Plugin) enableAutomute(userID string) (bool, error) {
+	return p.setAutomuteEnabledForUser(userID, true)
+}
+
+// disableAutomute mutes all of the user's linked channels and DMs and sets a preference to track that automute is
+// disabled. It assumes that the user is either not connected to MS Teams or has their primary platform set to MM.
+func (p *Plugin) disableAutomute(userID string) (bool, error) {
+	return p.setAutomuteEnabledForUser(userID, false)
+}
+
+func (p *Plugin) setAutomuteEnabledForUser(userID string, automuteEnabled bool) (bool, error) {
+	if channelsAutomuted := p.getAutomuteIsEnabledForUser(userID); channelsAutomuted == automuteEnabled {
+		// We've already automuted all the users' channels, so there's nothing to do
+		return false, nil
+	}
+
+	var membersToMute []*model.ChannelMemberIdentifier
+
+	channels, appErr := p.API.GetChannelsForTeamForUser("", userID, true)
+	if appErr != nil {
+		return false, errors.Wrap(appErr, fmt.Sprintf("Unable to get channels for user %s to automute them", userID))
+	}
+
+	for _, channel := range channels {
```

Comment:
I'm assuming this is simply unpaginated? Or does the server have some implicit pagination in play that might artificially limit the records returned?

Reply:
As far as I can tell, yeah, it's unpaged. At the very least, there's no paging mechanism I could use to handle if this was paged